packages/contracts/src/dollar/core/StakingShare.sol

```diff
@@ -134,15 +151,68 @@ contract StakingShare is ERC1155Ubiquity {
         return _stakes[id];
     }
 
+    /**
+     * @dev See {IERC1155-safeBatchTransferFrom}.
+     */
+    function safeBatchTransferFrom(
+        address from,
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    ) public virtual override(ERC1155, ERC1155Ubiquity) whenNotPaused {
+        super.safeBatchTransferFrom(from, to, ids, amounts, data);
+    }
+
+    function _burnBatch(
+        address account,
+        uint256[] memory ids,
+        uint256[] memory amounts
+    ) internal virtual override(ERC1155, ERC1155Ubiquity) whenNotPaused {
+        super._burnBatch(account, ids, amounts);
+    }
+
+    function _beforeTokenTransfer(
+        address operator,
+        address from,
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    ) internal virtual override(ERC1155, ERC1155Ubiquity) {
+        super._beforeTokenTransfer(operator, from, to, ids, amounts, data);
+    }
+
+    function uri(
+        uint256 tokenId
+    ) public view virtual override(ERC1155, ERC1155URIStorage) returns (string memory) {
+        return super.uri(tokenId);
```

Comment:
should we have a getter for _uri ?

Reply:
Not sure its important. @rndquu whats your opinion.

Reply:
If theres a question around it you should just add it. I think theres almost no downside to adding getters.

Reply:
@zgorizzo69 

This is how URI getter works: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol#L46

So if URI is empty for a particular token id then return this URI https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/ERC1155.sol#L30 else return this URI https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol#L22

So we don't need a separate getter for `ERC1155._uri` because we already have it here https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol#L46

Reply:
> should we have a getter for _uri ?

You mean can't we delete it? Both `ERC1155 .sol` and `ERC1155URIStorage.sol` have the `uri()` method so we must override it